src/infra/core/src/repos/metadata_block_repository_caching_inmem.rs

```diff
@@ -0,0 +1,181 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use async_trait::async_trait;
+use bytes::Bytes;
+use dashmap::DashMap;
+use kamu_core::{
+    ContainsBlockError,
+    ExternalTransferOpts,
+    GetBlockDataError,
+    GetBlockError,
+    GetBlockExternalUrlError,
+    GetBlockExternalUrlResult,
+    InsertBlockError,
+    InsertBlockResult,
+    InsertOpts,
+    MetadataBlockRepository,
+    ObjectRepository,
+};
+use opendatafabric::{MetadataBlock, Multihash};
+
+use crate::MetadataBlockRepositoryImpl;
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+struct MetadataBlockRepositoryCacheValue {
```

Comment:
If a type is local to module I would recommend sticking to short names like `CachedValue` - there is no danger of ambiguity or collision on module level.

Reply:
After [the hotfix](https://github.com/kamu-data/kamu-cli/pull/496), we don't need this structure -- we can just cache blocks, not bytes

So it's no longer actual